core/packages/PackageInfo.cc

```diff
@@ -57,4 +57,38 @@ string PackageInfo::show(const core::GlobalState &gs) const {
                          "::", [&](string *out, core::NameRef name) { absl::StrAppend(out, name.show(gs)); });
 }
 
+core::ClassOrModuleRef getTestSym(const core::GlobalState &gs) {
+    return core::Symbols::root().data(gs)->findMember(gs, core::Names::Constants::Test()).asClassOrModuleRef();
+}
+
+bool belongsToTestNamespace(const core::GlobalState &gs, const core::SymbolRef sym) {
+    auto testSym = getTestSym(gs);
+    if (!testSym.exists()) {
+        return false;
+    }
+
+    return sym.isUnderNamespace(gs, testSym);
+}
```

Comment:
Slight preference for inlining this function into `getRootSymbolForAutocorrectSearch` so that you only have to call `getTestSym` once, but meh.